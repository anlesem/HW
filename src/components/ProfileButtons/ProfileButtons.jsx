import style from './ProfileButtons.module.scss';

import Button from '@mui/material/Button';
import { Link } from 'react-router-dom';

export const ProfileButtons = () => {
  return (
    <div className={style.buttons}>
      <Link to="sign">
        <Button variant="outlined" data-testid={'profile-link-enter'}>
          Войти
        </Button>
      </Link>
      <Link to="sign/reg">
        <Button variant="outlined" data-testid={'profile-link-reg'}>
          Регистрация
        </Button>
      </Link>
    </div>
  );
};
