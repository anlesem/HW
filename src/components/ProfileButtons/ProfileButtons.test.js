import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { ProfileButtons } from './ProfileButtons';

describe('ProfileButtons', () => {
  it('Компонент существует', () => {
    expect(ProfileButtons).toBeInstanceOf(Function);
  });

  it('Снимок состояния', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <ProfileButtons />
      </BrowserRouter>
    );
    expect(asFragment(<ProfileButtons />)).toMatchSnapshot();
  });
});
