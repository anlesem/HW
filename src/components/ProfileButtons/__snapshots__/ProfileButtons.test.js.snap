// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileButtons Снимок состояния 1`] = `
<DocumentFragment>
  <div
    class="buttons"
  >
    <a
      href="/sign"
    >
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        data-testid="profile-link-enter"
        tabindex="0"
        type="button"
      >
        Войти
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </a>
    <a
      href="/sign/reg"
    >
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        data-testid="profile-link-reg"
        tabindex="0"
        type="button"
      >
        Регистрация
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </a>
  </div>
</DocumentFragment>
`;
